numpy/ma/core.py

```diff
@@ -3118,6 +3118,46 @@ def __array_wrap__(self, obj, context=None):
 
         return result
 
+    def __array_function__(self, function, types, args, kwargs):
+        """
+        Wrap numpy functions that have a masked implementation.
+
+        Parameters
+        ----------
+        function : callable
+            Numpy function being called
+        types : iterable of classes
+            Classes that provide an ``__array_function__`` override. Can
+            in principle be used to interact with other classes. Below,
+            mostly passed on to `~numpy.ndarray`, which can only interact
+            with subclasses.
+        args : tuple
+            Positional arguments provided in the function call.
+        kwargs : dict
+            Keyword arguments provided in the function call.
+        """
+        ma_func = np.ma.__dict__.get(function.__name__, function)
+        # Known incompatible functions
+        #   copy: np.copy() has subok semantics, which we can't pass through
+        #   unique: using ma.view(ndarray) internally doesn't work
+        incompatible_functions = {"copy", "unique"}
+        # The masked functions know how to work with ndarray and
+        # MaskedArrays, but can't guarantee anything about other types
+        handled_types = {np.ndarray, MaskedArray}
```

Comment:
I'm not sure if we want to allow subclasses or not, this is the safest for now... If we remove this limitation, then Astropy `MaskedColumn` which is a subclass of `MaskedArray` doesn't like the results.